packages/frontend-2/components/viewer/Controls.vue

```diff
@@ -364,28 +375,76 @@ const {
+  [ViewerKeyboardActions.ToggleDiscussions]: [[ModifierKeys.Shift], 't'],
+  [ViewerKeyboardActions.ToggleFilters]: [[ModifierKeys.Shift], 'f'],
+  [ViewerKeyboardActions.ToggleMeasurements]: [[ModifierKeys.Shift], 'r'],
+  [ViewerKeyboardActions.ToggleProjection]: [[ModifierKeys.Shift], 'p'],
+  [ViewerKeyboardActions.ToggleSectionBox]: [[ModifierKeys.Shift], 'b'],
+  [ViewerKeyboardActions.ZoomExtentsOrSelection]: [[ModifierKeys.Shift], 'space']
+}
+
+const getShortcutTitle = (action: ViewerKeyboardActions) =>
+  `(${getKeyboardShortcutTitle([...map[action][0], map[action][1]])})`
+
 const modelsShortcut = ref(
-  `Models (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 'm'])})`
+  `Models ${getShortcutTitle(ViewerKeyboardActions.ToggleModels)}`
 )
 const explorerShortcut = ref(
-  `Scene Explorer (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 'e'])})`
+  `Scene Explorer ${getShortcutTitle(ViewerKeyboardActions.ToggleExplorer)}`
 )
 const discussionsShortcut = ref(
-  `Discussions (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 't'])})`
+  `Discussions ${getShortcutTitle(ViewerKeyboardActions.ToggleDiscussions)}`
 )
 const zoomExtentsShortcut = ref(
-  `Fit to screen (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 'Space'])})`
+  `Fit to screen ${getShortcutTitle(ViewerKeyboardActions.ZoomExtentsOrSelection)}`
 )
 const projectionShortcut = ref(
-  `Projection (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 'p'])})`
+  `Projection ${getShortcutTitle(ViewerKeyboardActions.ToggleProjection)}`
 )
 const sectionBoxShortcut = ref(
-  `Section Box (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 'b'])})`
+  `Section Box ${getShortcutTitle(ViewerKeyboardActions.ToggleSectionBox)}`
 )
 const measureShortcut = ref(
-  `Measure Mode (${getKeyboardShortcutTitle([ModifierKeys.AltOrOpt, 'd'])})`
+  `Measure Mode ${getShortcutTitle(ViewerKeyboardActions.ToggleMeasurements)}`
 )
 
+const handleKeyboardAction = (action: ViewerKeyboardActions) => {
+  switch (action) {
+    case ViewerKeyboardActions.ToggleModels:
+      toggleActiveControl('models')
+      break
+    case ViewerKeyboardActions.ToggleExplorer:
+      toggleActiveControl('explorer')
+      break
+    case ViewerKeyboardActions.ToggleDiscussions:
+      toggleActiveControl('discussions')
+      break
+    case ViewerKeyboardActions.ToggleFilters:
+      toggleActiveControl('filters')
+      break
+    case ViewerKeyboardActions.ToggleMeasurements:
+      toggleMeasurements()
+      break
+    case ViewerKeyboardActions.ToggleProjection:
+      trackAndtoggleProjection()
+      break
+    case ViewerKeyboardActions.ToggleSectionBox:
+      toggleSectionBox()
+      break
+    case ViewerKeyboardActions.ZoomExtentsOrSelection:
+      trackAndzoomExtentsOrSelection()
+      break
+  }
+}
+
+Object.entries(map).forEach(([actionKey, [modifiers, key]]) => {
+  const action = Number(actionKey) as ViewerKeyboardActions
```

Comment:
@andrewwallacespeckle  If my assumption about numeric enums during runtime is correct, this doesn't resolve the issue. If `actionKey` is a string, I don't even know what you're gonna get out of `Number(string)`, maybe `NaN`? And then feeding that into the keyboard action handler, who know's whats happening.

In the forEach you should do `if (isString(actionKey)) return`